std/hash/_wasm/hash.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+import init, {
+  source,
+  create_hash as createHash,
+  update_hash as updateHash,
+  digest_hash as digestHash,
+  DenoHash,
+} from "./wasm.js";
+
+import * as hex from "../../encoding/hex.ts";
+import * as base64 from "../../encoding/base64.ts";
+import { Hasher, Message, OutputFormat } from "../hasher.ts";
+
+await init(source);
+
+const TYPE_ERROR_MSG = "hash: `data` is invalid type";
+
+export class Hash implements Hasher {
+  #hash: DenoHash;
+  #digested: boolean;
+
+  constructor(algorithm: string) {
```

Comment:
Should this be changed to: `algorithm: SupportedAlgorithm` instead of `algorithm: string`?

And you can remove `as string` 

https://github.com/denoland/deno/blob/b59bab30b2c06be8d14086ac202bd52e3ffb1fc2/std/hash/mod.ts#L32-L34

Reply:
In order to use `SupportedAlgorithm` in `hash.ts`, I have to import `mod.ts`.
But `mod.ts` is also importing `hash.ts`, so I thought it would make cycle.
Do you think it would be okay?

Reply:
In architectural point of view, I think `Hash` class doesn't need to know about `SupportedAlgorithm` since it is just an interface for WASM.
`SupportedAlgorithm` would do enough of its job for guarding algorithm for `createHash` defined in `mod.ts`.